code/game/gamemodes/miniantags/pulsedemon/pulsedemon.dm

```diff
@@ -0,0 +1,823 @@
+
+/mob/living/simple_animal/demon/pulse_demon/proc/gen_speech_name()
+	. = ""
+	for(var/i = 1 to 10)
+		. += pick("!", "@", "#", "$", "%", "^", "&", "*")
+
+/mob/living/simple_animal/demon/pulse_demon/say(message, verb, sanitize = TRUE, ignore_speech_problems = FALSE, ignore_atmospherics = FALSE, ignore_languages = FALSE)
+	if(client && check_mute(client.ckey, MUTE_IC))
+		to_chat(src, "<span class='danger'>You cannot speak in IC (Muted).</span>")
+		return FALSE
+
+	if(sanitize)
+		message = trim_strip_html_properly(message)
+
+	if(stat)
+		if(stat == DEAD)
+			return say_dead(message)
+		return FALSE
+
+	if(current_robot)
+		var/turf/T = get_turf(src)
+		log_say("[key_name_admin(src)] (@[T.x], [T.y], [T.z]) made [current_robot]([key_name_admin(current_robot)]) say: [message]")
+		log_admin("[key_name_admin(src)] made [key_name_admin(current_robot)] say: [message]")
+		message_admins("<span class='notice'>[key_name_admin(src)] made [key_name_admin(current_robot)] say: [message]</span>")
+		// don't sanitize again
+		current_robot.say(message, null, FALSE, ignore_speech_problems, ignore_atmospherics, ignore_languages)
+		return TRUE
+
+	var/message_mode = parse_message_mode(message, "headset")
+
+	if(message_mode)
+		if(message_mode == "headset")
+			message = copytext(message, 2)
+		else
+			message = copytext(message, 3)
+
+	message = trim_left(message)
+
+	var/list/message_pieces = list()
+	if(ignore_languages)
+		message_pieces = message_to_multilingual(message)
+	else
+		message_pieces = parse_languages(message)
+
+	// hivemind languages
+	if(istype(message_pieces, /datum/multilingual_say_piece))
+		var/datum/multilingual_say_piece/S = message_pieces
+		S.speaking.broadcast(src, S.message)
+		return TRUE
+
+	if(!LAZYLEN(message_pieces))
+		. = FALSE
+		CRASH("Message failed to generate pieces. [message] - [json_encode(message_pieces)]")
+
+	create_log(SAY_LOG, "[message_mode ? "([message_mode])" : ""] '[message]'")
+
+	playsound(get_turf(src), pick(speech_sounds), 30, TRUE)
+	if(istype(loc, /obj/item/radio))
+		var/obj/item/radio/R = loc
+		name = gen_speech_name()
+		R.talk_into(src, message_pieces, message_mode, verbage = verb)
+		name = real_name
+		return TRUE
+	else if(istype(loc, /obj/machinery/hologram/holopad))
+		var/obj/machinery/hologram/holopad/H = loc
+		name = "[H]"
+		for(var/mob/M in get_mobs_in_view(7, H))
+			M.hear_say(message_pieces, verb, FALSE, src)
+		name = real_name
+		return TRUE
+
+	emote("me", message = "[pick(emote_hear)]")
+	return TRUE
+
+/mob/living/simple_animal/demon/pulse_demon/update_runechat_msg_location()
+	if(istype(loc, /obj/machinery/hologram/holopad))
+		runechat_msg_location = loc.UID()
+	else
+		return ..()
+
+/mob/living/simple_animal/demon/pulse_demon/visible_message(message, self_message, blind_message)
+	// overriden because pulse demon is quite often in non-turf locs, and /mob/visible_message acts differently there
+	for(var/mob/M in get_mobs_in_view(7, src))
```

Comment:
does this properly account for screen width?

Reply:
No

Reply:
> please fix it then

I'd need to bump it up to 9, which would cause even more issues as you could now get the message while the pulse demon is off screen